### DISCUSSION TITLE
Fist impressions feedback

### DISCUSSION THREAD

**Discussion by BKG01** [zed-industries/zed#10820] [Feature Requests]
I just set up Zed yesterday,(been on my to do for a bit), and it's a great first impression. It seemed to have typescript errors, eslint and prettier working without doing anything really. (I just have those configs in the repo).  

Some vim tweaks made and then that was pretty much all I needed.  Definitely the quickest to get setup on any editor I've tried. 
Very smooth editor experience too.

It's amazing how many features are in already and working well.  
There are just a couple I'm missing from the day to day which I'll mention in case it's useful.

- Turning off tabs (Ideally a user wouldn't have to think about what's open or not, just what is saved or not and I use auto save here too to remove that thought. Best to let the editor manage open tab memory and close as needed and just let the user control how many open panes they want. Preview tabs is a good initial step forward)
- Git gutter click to preview changes (I had no idea how useful this was until it was missing.... I seem to use this a lot more than I realised)
- Git diffing
- Debugger (with breakpoint on exceptions, step over, step into etc... Not a day to day, but really useful)

However I haven't really discovered the benefits of having a tree-sitter based editor yet and I'm still editing like before. (It's only been a day so I might have missed this stuff so far).
I'm still editing with vim like it's text instead of ideally selecting and changing the parts of an AST (which I thought would be a key benefit of Zed with tree-sitter). 
I would love it if there was a kakoune style select, then action that ideally worked on sections of the AST so that....

Z(ed) = select(AST) + action.    

to edit code....

maybe that's a bit idealistic, but I'd love to know how to utilise the benefits of tree sitter better. 

Thanks. :)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
